ql/src/java/org/apache/hadoop/hive/ql/io/AcidUtils.java

```diff
@@ -3243,6 +3244,21 @@ public static AcidDirectory getAcidStateFromCache(Supplier<FileSystem> fileSyste
     return value.getDirInfo();
   }
 
+  public static void tryInvalidateDirCache(org.apache.hadoop.hive.metastore.api.Table table) {
+    if (dirCacheInited.get()) {
+      TableName tableName = new TableName(table.getCatName(), table.getDbName(), table.getTableName());
+      String key = tableName.getNotEmptyDbTable() + "_" + table.getSd().getLocation();
```

Comment:
I think you can use here AcidUtils.getFullTableName(table.getDbName(), table.getTableName())

Reply:
Done